src/images/falseColourPanel.component.tsx

```diff
@@ -55,6 +63,111 @@ interface FalseColourPanelProps extends FalseColourParams {
+  colourMap: string,
+  handleColourMapChange: (event: SelectChangeEvent) => void,
+  mainColourMap: ColourMapsParams
+) => {
+  const mainColourMapTypeName = Object.keys(mainColourMap)[0];
+  const mainColourMapNames = Object.values(mainColourMap)[0];
+
+  return (
+    <Select
+      labelId="colour-map-select-label"
+      id="colour-map-select"
+      value={colourMap}
+      label="Colour Map"
+      onChange={handleColourMapChange}
+    >
+      <MenuItem value="">
+        <em>Default</em>
+      </MenuItem>
+
+      <MyListSubheader>{mainColourMapTypeName}</MyListSubheader>
+      {mainColourMapNames.map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+    </Select>
+  );
+};
+
+const ColourMapSelectExtended = (
+  colourMap: string,
+  handleColourMapChange: (event: SelectChangeEvent) => void,
+  extendedColourMapNames: ColourMapsParams
+) => {
+  const mainColourMapTypeNames = Object.keys(extendedColourMapNames);
+  const mainColourMapNames = Object.values(extendedColourMapNames);
+  return (
+    <Select
+      labelId="colour-map-select-label"
+      id="colour-map-select"
+      value={colourMap}
+      label="Colour Map"
+      onChange={handleColourMapChange}
+    >
+      <MenuItem value="">
+        <em>Default</em>
+      </MenuItem>
+
+      <MyListSubheader>{mainColourMapTypeNames[0]}</MyListSubheader>
+      {mainColourMapNames[0].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[1]}</MyListSubheader>
+      {mainColourMapNames[1].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[2]}</MyListSubheader>
+      {mainColourMapNames[2].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[3]}</MyListSubheader>
+      {mainColourMapNames[3].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[4]}</MyListSubheader>
+      {mainColourMapNames[4].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[5]}</MyListSubheader>
+      {mainColourMapNames[5].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
+
+      <MyListSubheader>{mainColourMapTypeNames[6]}</MyListSubheader>
+      {mainColourMapNames[6].map((colourMap) => (
+        <MenuItem key={colourMap} value={colourMap}>
+          {colourMap}
+        </MenuItem>
+      ))}
```

Comment:
Could this code not be achieved with a loop over `mainColourMapTypeNames`/`mainColourMapNames` to reduce code repetition? Or just a simple for loop, and access both arrays by index.

Reply:
Also, if this is then done by a loop, then there wouldn't need to be two separate components for `ColourMapMain` and `ColourMapExtended` - the difference between them would just be that the main one is only supplied with one item to "loop" over

Reply:
The menuItems do not work correctly when rendered in a loop. When I initially did it a loop the menuItems were not clickable.

Reply:
The issue is you need to ensure that the `MenuItem`s are direct children of the `Select`, which can be done like so:

```
{mainColourMapNames.map((mapNames, index) => {
        return [
          <MyListSubheader>{mainColourMapTypeNames[index]}</MyListSubheader>,
          mapNames.map((colourMap) => (
            <MenuItem key={colourMap} value={colourMap}>
              {colourMap}
            </MenuItem>
          )),
        ];
      })}
```

Here, we ensure we're returning arrays, which React parses to make every item a direct descendant.

Reply:
Actually I've got it working using a loop. The Select component was complaining about me using react.Fragment before so I've just implemented a loop without using React.fragment